### PR TITLE
[google_identity_services_web] Migrate all types to JSObject

### PR DESCRIPTION
**Pull request by kevmoo**
And cleanup other legacy JS-interop
Also update tests so they pass with WebAssembly

### CODE REVIEW

File: packages/google_identity_services_web/example/integration_test/utils.dart [hunk 1]
```diff
@@ -33,6 +49,12 @@ ExpectConfigValueFn createExpectConfigValue(JSObject config) {
 Matcher isAJs(String thing) => isA<JSAny?>()
     .having((JSAny? p0) => p0.typeofEquals(thing), 'typeof "$thing"', isTrue);
 
+final Matcher isJSArray = isA<JSAny?>()
+    .having((JSAny? p0) => _isArray(p0).toDart, 'Array.isArray', isTrue);
```
Comment (srujzs) [inline packages/google_identity_services_web/example/integration_test/utils.dart:53]: It's up to you if you'd rather do an `isA<JSArray>` (`isA` being from `dart:js_interop` and not `package:test`) or what you have here. It probably shouldn't matter but `Array.isArray` is a little stricter.

**Review by srujzs — APPROVED** [review]
Thanks again for doing this Kevin!

**Comment by kevmoo** [comment]
We have dart functions in these types...that probably should be JSFunction, right?

**Comment by srujzs** [comment]
> We have dart functions in these types...that probably should be JSFunction, right?

Can you share an example of what you're referring to?

**Comment by ditman** [comment]
Which reminds me that we need to run our web integration tests with `--wasm` enabled! I'll merge this ASAP.

**Review by ditman — APPROVED** [review]
LGTM!

File: packages/google_identity_services_web/CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 ## 0.3.3
 
-* Move all the JavaScript types to extend `JSObject`.
+* Migrates all the JavaScript types to the latest syntax.
```
Comment (kevmoo) [inline packages/google_identity_services_web/CHANGELOG.md:3]: Yeah, but that doesn't tell the user WHAT changed. The fact that these types are now `JSObject` *is* the user-visible change.

Comment (ditman) [reply packages/google_identity_services_web/CHANGELOG.md:3]: Users never see these objects, they don't care about the underlying representation, but let me revert.

**Pull request merged**